src/test/kotlin/no/nav/personbruker/dittnav/eventaggregator/done/DoneEventServiceTest.kt

```diff
@@ -0,0 +1,260 @@
+                service.processEvents(records)
+            }
+        } `should throw` UntransformableRecordException::class
+
+        capturedNumberOfBeskjedEntitiesWrittenToTheDb.captured.size `should be` 2
+
+        coVerify(exactly = 1) { repository.writeDoneEventsForBeskjedToCache(any()) }
+        coVerify(exactly = 1) { repository.writeDoneEventToCache(emptyList()) }
+        coVerify(exactly = 1) { repository.writeDoneEventsForInnboksToCache(emptyList()) }
+        coVerify(exactly = 1) { repository.writeDoneEventsForOppgaveToCache(emptyList()) }
+    }
+
+    @Test
+    fun `skal forkaste eventer som har valideringsfeil`() {
+        val tooLongFodselsnr = "1".repeat(12)
+        val beskjedInDbToMatch = BrukernotifikasjonObjectMother.giveMeBeskjed(tooLongFodselsnr)
+        val records = createMatchingRecords(beskjedInDbToMatch)
+
+        coEvery {
+            repository.fetchActiveBrukernotifikasjonerFromView()
+        } returns listOf(beskjedInDbToMatch)
+
+        runBlocking {
+            service.processEvents(records)
+        }
+
+        coVerify(exactly = 1) { repository.writeDoneEventToCache(emptyList()) }
+        coVerify(exactly = 1) { repository.writeDoneEventsForBeskjedToCache(emptyList()) }
+        coVerify(exactly = 1) { repository.writeDoneEventsForInnboksToCache(emptyList()) }
+        coVerify(exactly = 1) { repository.writeDoneEventsForOppgaveToCache(emptyList()) }
+    }
+
+    private fun createMatchingRecords(entitiesInDbToMatch: List<Brukernotifikasjon>): ConsumerRecords<Nokkel, no.nav.brukernotifikasjon.schemas.Done> {
+        val listOfConsumerRecord = mutableListOf<ConsumerRecord<Nokkel, no.nav.brukernotifikasjon.schemas.Done>>()
+        entitiesInDbToMatch.forEach { entity ->
+            val done = AvroDoneObjectMother.createDoneRecord(entity.eventId, entity.fodselsnummer)
+            listOfConsumerRecord.add(done)
+        }
+        return ConsumerRecordsObjectMother.giveMeConsumerRecordsWithThisConsumerRecord(listOfConsumerRecord)
+    }
+
+    private fun createMatchingRecords(entityInDbToMatch: Brukernotifikasjon): ConsumerRecords<Nokkel, no.nav.brukernotifikasjon.schemas.Done> {
+        val matchingDoneEvent = AvroDoneObjectMother.createDoneRecord(entityInDbToMatch.eventId, entityInDbToMatch.fodselsnummer)
+        return ConsumerRecordsObjectMother.giveMeConsumerRecordsWithThisConsumerRecord(matchingDoneEvent)
+    }
+
+    private fun captureNumberOfDoneEntitiesWrittenToTheDb(): CapturingSlot<List<Done>> {
+        val capturedNumberOfDoneWrittenToTheDb = slot<List<Done>>()
+        coEvery { repository.writeDoneEventToCache(capture(capturedNumberOfDoneWrittenToTheDb)) } returns Unit
+        return capturedNumberOfDoneWrittenToTheDb
+    }
+
+    private fun captureBeskjedEventsWrittenToDb(): CapturingSlot<List<Done>> {
+        val numberOfEntities = slot<List<Done>>()
+        coEvery { repository.writeDoneEventsForBeskjedToCache(capture(numberOfEntities)) } returns Unit
```

Comment:
Kunne man ha sendt inn repository.writeDoneE.....-funksjonen, og på den måten klart seg med èn sånn captureEvents-funksjon? Eller blir det tungvindt med capture?

Reply:
Jeg er helt enig i poenget, men er redd lesbarheten blir dårligere. Men jeg skal ta en ekstra titt, og se om det lar seg gjøre på en lesbarmåte.

Reply:
Jeg inline-et bare de hjelpemetodene, fordi det ble minst kode. Trodde først at jeg kom til å bruke mange av de i samme test, men ser nå at jeg kun bruke de en eller to ganger hver totalt.

Reply:
Synes det ble veldig bra :)